plugins/MLTModule.cpp

```diff
@@ -103,6 +103,43 @@ MLTModule::init(std::shared_ptr<appfwk::ModuleConfiguration> mcfg)
 //   ci.add(i);
 // }
 
+void
+MLTModule::generate_opmon_data()
+{
+  opmon::ModuleLevelTriggerInfo info;
+
+  info.set_td_msg_received_count( m_td_msg_received_count.load() );
+  info.set_td_sent_count( m_td_sent_count.load() );
+  info.set_td_inhibited_count( m_td_inhibited_count.load() );
+  info.set_td_paused_count( m_td_paused_count.load() );
+  info.set_td_queue_timeout_expired_err_count( m_td_queue_timeout_expired_err_count.load() );
+  info.set_td_total_count( m_td_total_count.load() );
+
+  std::lock_guard<std::mutex>	guard(m_trigger_mutex);
+  for ( auto & [type, counts] : m_trigger_counters ) {
+    opmon::TriggerDecisionInfo td_info;
+    td_info.set_received(counts.received.exchange(0));
+    td_info.set_sent(counts.sent.exchange(0));
+    td_info.set_failed_send(counts.failed_send.exchange(0));
+    td_info.set_paused(counts.paused.exchange(0));
+    td_info.set_inhibited(counts.inhibited.exchange(0));
+    auto name = dunedaq::trgdataformats::get_trigger_candidate_type_names()[type];
+    publish( std::move(td_info), {{"type", name}} );
```

Comment:
The time of calling publish should be as close as possible to the filling of the structure. So please move the block on the trigger types before the creation of `opmon::ModuleLevelTriggerInfo` (but in that case use a unique_lock instead of a lock gard and remember to unlock it) or put the block after you publish `opmon::ModuleLevelTriggerInfo`